pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,105 +38,62 @@ type Terminator struct {
 	coreV1Client  corev1.CoreV1Interface
 }
 
-// cordonNodes takes in a list of expired nodes as input and cordons them
-func (t *Terminator) cordonNodes(ctx context.Context) error {
-	// 1. Get terminable nodes
-	nodeList, err := t.getLabeledNodes(ctx, provisioning.ProvisionerTerminablePhase)
-	if err != nil {
-		return err
-	}
-	// 2. Cordon nodes
-	for _, node := range nodeList {
-		persisted := node.DeepCopy()
-		node.Spec.Unschedulable = true
-		node.Labels = functional.UnionStringMaps(
-			node.Labels,
-			map[string]string{provisioning.ProvisionerPhaseLabel: provisioning.ProvisionerDrainingPhase},
-		)
-		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
-		}
-		zap.S().Debugf("Cordoned node %s", node.Name)
+// cordonNode cordons a node
+func (t *Terminator) cordonNode(ctx context.Context, node *v1.Node) error {
+	persisted := node.DeepCopy()
+	node.Spec.Unschedulable = true
+	if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+		return fmt.Errorf("patching node %s, %w", node.Name, err)
 	}
+	zap.S().Debugf("Cordoned node %s", node.Name)
 	return nil
 }
 
-// terminateNodes takes in a list of expired non-drained nodes and calls drain on them
-func (t *Terminator) terminateNodes(ctx context.Context) error {
-	// 1. Get draining nodes
-	draining, err := t.getLabeledNodes(ctx, provisioning.ProvisionerDrainingPhase)
+// drainNode evicts pods from the node and returns true when fully drained
+func (t *Terminator) drainNode(ctx context.Context, node *v1.Node) (bool, error) {
```

Comment:
do you need the bool?  Couldn't you just define a successful drain as not returning an error?

Reply:
Cases:
- (true, nil)
 - Fully drained, good to terminate
- (false, nil)
  - Still draining, but nothing failing here (has pod with `do-not-evict` or just hasn't completed yet)
  - Need to include a bool here for this case
- (false, err)
  - still draining, but something failed

Reply:
I see, so since this is async, we need the bool to determine if it's still working is the tldr.  So on the first termination run, we shouldn't ever finish if there is at least one non-daemonset pod on the node being drained.  I guess that's not an issue now since we only drain if the node is empty.  

But anyways, this seems fine, at lest for now. Maybe there's an optimization later to recheck after the eviction starts in the same reconcile call rather than waiting on the interval.